### ISSUE TITLE
Blur "NSFW" Images within UI

### ISSUE THREAD

**Issue by kinglunalilo** [mcmonkeyprojects/SwarmUI#488]
### Feature Idea

1)within the thumbnails among LORAS and MODELS

2)Generated images themselves that are created in output

Optional: An "AUTO" blur feature would be a miracle.  Reveals on click?  Not sure how that would work if trying to add a LORA to project.

**Comment by mcmonkey4eva** [comment]
Considering swarm is local/personal, can't you just, yknow, not generate nsfw things?

**Comment by iwr-redmond** [comment]
To be fair, some checkpoints have their phone off the hook.

**Comment by klromans557** [comment]
> Considering swarm is local/personal, can't you just, yknow, not generate nsfw things?

Yeah, but sometimes I wanna show my homie how the magic works without all the big-tiddie-model thumbnails showing. Even an option to switch all model thumbnails to the default background with a single toggle would do the trick. 
I agree with you that blurring generated images would probably not be worth the effort and not making them is the way to go.

**Comment by vvirtues** [comment]
> > Considering swarm is local/personal, can't you just, yknow, not generate nsfw things?
> 
> Yeah, but sometimes I wanna show my homie how the magic works without all the big-tiddie-model thumbnails showing. Even an option to switch all model thumbnails to the default background with a single toggle would do the trick. I agree with you that blurring generated images would probably not be worth the effort and not making them is the way to go.

Or use a list view with no thumbnails would be nice

**Comment by iwr-redmond** [comment]
According to the [updated BFL license](https://bfl.ai/legal/self-hosted-commercial-license-terms):

> You must [...] implement and maintain content filtering measures for your use of the Licensed FLUX [dev] Model or permitted Derivatives to prevent the creation, display, transmission, generation, or dissemination of unlawful or infringing content.
>
> [...]
>
> We reserve the right to require you to use content filtering and content provenence controls we provide or endorse.

Huggingface is trying to [offset](https://github.com/huggingface/diffusers/blob/d7dd924ece56cddf261cd8b9dd901cbfa594c62c/docs/source/en/api/pipelines/flux.md#kontext) its potential liability to downstream Diffusers integrators:

> Flux Kontext comes with an integrity safety checker, which should be run after the image generation step. To run the safety checker, install the official repository from [black-forest-labs/flux](https://github.com/black-forest-labs/flux).

I reckon the multiuser nature of SwarmUI may put the software in the firing line, although it's probably too early to tell for sure just yet. However, SwarmUI won't be able to follow HF as it is directly end-user facing.

**Comment by mcmonkey4eva** [comment]
Swarm is intended primarily for direct personal use at home. The safety checker is the human operating the interface. Any software checks that run direct on your own computer are redundant and easily bypassed by anyone that wants to.

**Comment by iwr-redmond** [comment]
I suppose that puts SwarmUI into option 2, which I, perhaps foolishly!, omitted for brevity:

> [...] or (ii) ensure Output undergo review for unlawful or infringing content before public or non-public distribution, display, transmission or dissemination.

There is nothing on the plain reading to suggest that this review could not be a manual one, and the onus for such review put onto end users. FooocusPlus has some [language in its Readme](https://github.com/DavidDragonsage/FooocusPlus/blob/3b764e7f076d46044b903260598933d3905137cc/readme.md#licensing) that may be adoptable here:

> You also agree that you will not use FooocusPlus to:
>
> [...] 
>
> - Violate the license agreements of included software packages or AI models

**Comment by mcmonkey4eva** [comment]
Yes, Swarm already has had that for a long time. This is the first screen you see after launching swarm the first time:

![Image](https://github.com/user-attachments/assets/53ae6d4d-6ea5-4e10-b688-3bdf9fc735d9)

**Comment by iwr-redmond** [comment]
Tells you how often I read the fine print! 😊

**Comment by bman654** [comment]
also it is only in the firing line when someone decides to make a publicly available generation site powered by SwarmUI.  And then the burden would be on that person to make the site compliant, not on the SwarmUI project itself.

**Comment by iwr-redmond** [comment]
BFL are apparently still [rewriting the license](https://www.reddit.com/r/StableDiffusion/comments/1llytk0/flux_license_updated_on_github_today_to_clarify/) based on initial feedback. Not good that they didn't road test their legals first, decent that they are fixing issues that have come up subsequently. I ultimately agree with the contention that trying to tie local code in knots is ineffective, and hope that whatever legal kinks remain can be sorted out in time at the BFL end.
